gandalf/src/main/java/io/github/btkelly/gandalf/Gandalf.java

```diff
@@ -46,8 +68,56 @@ private static Gandalf createInstance() {
         return new Gandalf();
     }
 
-    public String getBootstrapUrl() {
-        return bootstrapUrl;
+    /**
+     * Saves the provided {@link OptionalUpdate} to the history checker.
+     * @param optionalUpdate the provided current optional update information
+     * @return {@code true} if {@code optionalUpdate} was successfully saved
+     */
+    public boolean save(@NonNull final OptionalUpdate optionalUpdate) {
+        return this.historyChecker.save(optionalUpdate);
+    }
+
+    /**
+     * Saves the provided {@link Alert} to the history checker.
+     * @param alert this should be the current alert
+     * @return {@code true} if {@code alert} was successfully saved
+     */
+    public boolean save(@NonNull final Alert alert) {
+        return this.historyChecker.save(alert);
+    }
+
+    /**
+     * Starts the flow on checking a remote file and determining if any updates or alerts are available.
+     * @param gandalfCallback - a callback interface to respond to events from the bootstrap check
+     */
+    public void shallIPass(final GandalfCallback gandalfCallback) {
+        new BootstrapApi(context, bootstrapUrl)
+                .fetchBootstrap(new Callback() {
+                    @Override
+                    public void onFailure(Call call, IOException e) {
+                        //In any error case we should let the user in as to not block based on a bug
+                        gandalfCallback.onNoActionRequired();
+                    }
+
+                    @Override
+                    public void onResponse(Call call, Response response) throws IOException {
+                        Gson gson = new GsonBuilder()
```

Comment:
should we add a todo to put this into some sort of getter method, for in the future when we have custom deserializers